src/connection/database/login_sign_up_tabs.jsx

```diff
@@ -38,9 +42,16 @@ export default class LoginSignUpTabs extends React.Component {
   }
 
   handleSignUpClick() {
+    if (this.props.signUpLink) {
```

Comment:
I don't think so, in this context there are 2 scenarios:

regular webapp: the redirection will change the entire page. It is irrelevant if lock hides or not
spa: changing the route will not force a redirection so lock keeps visible unless it is explicitly hidden (this change).

While implementing lock, you might want it to not be closable (forcing signup or login on a page?) but you will want it to hide if the user is redirected to a different signup page.